tycho-integration-test/src/main.rs

```diff
@@ -160,669 +126,301 @@ async fn run(cli: Cli) -> miette::Result<()> {
         .into_diagnostic()
         .wrap_err("Failed to connect to provider")?;
 
-    // Init stream
-    let mut pairs: HashMap<String, ProtocolComponent> = HashMap::new();
-    let mut protocol_stream = build_protocol_stream(&cli, chain, &tycho_url, &all_tokens).await?;
-    let mut first_message_skipped = false;
+    // Load tokens from Tycho
+    let tycho_url =
+        get_default_url(&chain).ok_or_else(|| miette!("No default Tycho URL for chain {chain}"))?;
+    info!(%tycho_url, "Loading tokens...");
+    let all_tokens =
+        load_all_tokens(&tycho_url, false, Some(cli.tycho_api_key.as_str()), chain, None, None)
+            .await;
+    info!(%tycho_url, "Loaded tokens");
 
-    while let Some(res) = protocol_stream.next().await {
-        let now = std::time::SystemTime::now()
-            .duration_since(std::time::UNIX_EPOCH)
-            .unwrap()
-            .as_secs();
+    // Run streams in background tasks
+    let (tx, mut rx) = tokio::sync::mpsc::channel(64);
+    if let Ok(protocol_stream_processor) = ProtocolStreamProcessor::new(
+        chain,
+        tycho_url.clone(),
+        cli.tycho_api_key.clone(),
+        cli.tvl_threshold,
+    ) {
+        protocol_stream_processor
+            .run_stream(&all_tokens, tx.clone())
+            .await?;
+    }
+    if let Ok(rfq_stream_processor) =
+        RfqStreamProcessor::new(chain, cli.tvl_threshold, cli.max_n_simulations)
```

Comment:
What happens if one of them fails? Are we throwing away the error? Shouldn't we bubble it up instead?